scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,227 @@
+extern crate core;
+
+use anyhow::{anyhow, bail, Error};
+use itertools::{chain, Itertools};
+use regex::Regex;
+use std::path::{Path, PathBuf};
+
+fn main() -> anyhow::Result<(), Error> {
+    let text_w_anchors = search_for_patterns_in_project("ANCHOR")?;
+
+    let (starts, ends) = extract_starts_and_ends(&text_w_anchors)?;
+
+    let (valid_anchors, anchor_errors) = filter_valid_anchors(starts, ends);
+
+    let text_mentioning_include = search_for_patterns_in_project("{{#include")?;
+    let includes = parse_includes(text_mentioning_include)?;
+
+    let (include_errors, additional_warnings) = validate_includes(includes, valid_anchors);
+
+    report_warnings(&additional_warnings);
+
+    if !anchor_errors.is_empty() || !include_errors.is_empty() {
+        report_errors("anchors", &anchor_errors);
+        report_errors("includes", &include_errors);
+        panic!("Finished with errors");
+    }
+    Ok(())
+}
+
+fn report_errors(error_type: &str, errors: &[Error]) {
+    eprintln!("Invalid {} detected!", error_type);
+    for error in errors {
+        eprintln!("{error}")
+    }
+}
+
+fn report_warnings(warnings: &[Error]) {
+    for warning in warnings {
+        eprintln!("WARNING! {warning}")
+    }
+}
+
+fn validate_includes(
+    includes: Vec<Include>,
+    valid_anchors: Vec<Anchor>,
+) -> (Vec<Error>, Vec<Error>) {
+    let (pairs, errors): (Vec<_>, Vec<_>) = includes
+        .into_iter()
+        .map(|include| {
+            let mut maybe_anchor = valid_anchors.iter().find(|anchor| {
+                anchor.file == include.anchor_file && anchor.name == include.anchor_name
+            });
+
+            match maybe_anchor.take() {
+                Some(anchor) => Ok((include, anchor.clone())),
```

Comment:
It would seem you're not actually using the first part of the pair. Perhaps just collect used anchors?